src/background/Wallet/Wallet.ts

```diff
@@ -1704,12 +1711,13 @@ class PublicController {
     try {
       const chain = networks.getChainById(chainId);
       // Switch immediately and return success
-      this.wallet.setChainForOrigin(chain, origin);
+      this.wallet.setChainForOrigin({ chain, origin });
       // return null in next tick to give provider enough time to change chainId property
       return new Promise((resolve) => {
         setTimeout(() => resolve(null));
       });
     } catch (e) {
+      this.wallet.emitter.emit('switchChainError', chainId, origin);
```

Comment:
it looks like you can call `emitter` directly here.

Reply:
It will actually fix this case :)

Reply:
✅